src/somef/extract_software_type.py

```diff
@@ -0,0 +1,309 @@
+            else:
+                total_files+=1
+            if file.endswith(script_extensions):
+                bad_extensions=True
+
+            if file.endswith((".ipynb", ".rmd",'.Rmd',".jl")):
+                notebook_path = os.path.join(root, file)
+                try:
+                    has_code = False
+                    num_code_cells = 0
+                    num_total_cells = 0
+
+                    if file.endswith(".ipynb"):
+                        nb = nbformat.read(notebook_path, as_version=4)
+                        for cell in nb['cells']:
+                            if cell['cell_type'] == 'code':
+                                num_total_cells += 1
+                                if cell['source'].strip():
+                                    num_code_cells += 1
+                                    has_code = True
+                        if has_code and num_code_cells / num_total_cells >= 0.55:
+                            code_notebooks += 1
+                    elif file.endswith(".rmd") or file.endswith('.Rmd'):
+                        code_notebooks += 1
+                    elif file.endswith('.jl'):
+                        code_notebooks +=1
+
+
+                    total_notebooks += 1
+
+                except Exception as e:
+                    print(f"Error reading notebook file {notebook_path}: {str(e)}")
+                    pass
+
+    if code_notebooks/total_notebooks>=0.5 and bad_extensions==False:
+        return True
+    else:
+        return False
+
+
+def check_ontologies(path_repo):
+    """Function which detects if repository is an Ontology based on files present
+       and the non-existence of script files"""
+    extensions = set()
+    onto_extensions=set()
+    extensions_accepted = [".iml",".gitignore",".md",".txt", ".pdf",  ".xlsx",".xlx",  ".csv",  ".tsv", ".yaml",".yml",  ".jpg",   ".png",   ".gif",   ".wav",   ".mp3",   ".avi",   ".tiff", ".svg",".properties",".css",".html",".xml",".js",".json",".rst",".ttl","",".woff",".woff2",".scss",".prettierrc",".lock",".jsonld",".dot",".graphml",".drawio",".ttl",".nt",".owl",".htaccess",".conf",".cfg",".owl2",".nq",".n3",".trig",".rdf",".icon"]
+    archive_extensions = [".zip", ".tar", ".tar.gz", ".tar.bz2", ".tar.xz", ".tar.Z", ".rar", ".7z", ".gz", ".bz2", ".xz", ".Z"]
+    for root, dirs, files in os.walk(path_repo):
+        for file in files:
+            _, ext = os.path.splitext(file)
+            if ext in [".ttl",".nt",".owl",".htaccess",".conf",".cfg",".owl2",".nq",".n3",".trig",".rdf"]:
+                onto_extensions.add(ext)
+            else:
+                extensions.add(ext)
+    for element in list(extensions):
+        if element not in extensions_accepted or (element in archive_extensions and onto_extensions is None):
+            print("here")
+            return False
+    if not any(ext not in list(onto_extensions) for ext in [".ttl",".nt", ".owl", ".rdf", ".owl2"]):
+        print("here2")
```

Comment:
what is this?

Reply:
this checks if the repository contains any ontology files inside and if not it will automatically return False

Reply:
ok, although the `ontologies` field in the dictionary also does this